.github/workflows/release.yaml

```diff
@@ -114,6 +140,21 @@ jobs:
     secrets:
       registry-password: ${{ secrets.CR_PAT }}
 
+  generate-background-controller-provenance:
+    needs: release-images
+    permissions:
+      id-token: write   # To sign the provenance.
+      packages: write   # To upload assets to release.
+      actions: read     # To read the workflow path.
+    # NOTE: The container generator workflow is not officially released as GA.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@v1.4.0
+    with:
+      image: ghcr.io/${{ github.repository_owner }}/background-controller
+      digest: "${{ needs.release-background-controller.outputs.background-controller-digest }}"
```

Comment:
We used to create these digests from the `.github/workflow/reuse.yaml`, and I just realized it was removed in https://github.com/kyverno/kyverno/commit/7176e7282b52e3e646c6ed9d2b9340a9f460fad3#diff-495a82cf3f64b0a2d7d6bbd34baa976d5a19f6eaa5fa8309a515e50e74563ded.

This will cause the failure when generating provenance for our images during the release, @eddycharly - why did we remove it?

Reply:
And I wonder what's the best way to prevent this happening again🤔

Reply:
@realshuting this was refactored in smaller and more modular actions.
what is going to fail here ?

Reply:
It should come from the `release-images` job, looks like it won't work as is.

Reply:
This is how we used to build digests https://github.com/kyverno/kyverno/blob/af3882f869c1fb36cbed76564f048157bd11b3cd/.github/workflows/reuse.yaml#L22-L34.

Reply:
The digest is now an output of the `publish-image` action https://github.com/kyverno/kyverno/blob/16e870798a8e6300c0c2d59f78d030e428290f02/.github/actions/publish-image/action.yaml#L37-L40

Reply:
Thanks!

Is this the correct way of referencing the specific digest for a controller?
```
digest: "${{ needs.release-background-controller.outputs.background-controller-digest }}"
```

Reply:
Thanks!

Is this the correct way of referencing the specific digest for a controller?
```
digest: "${{ needs.release-background-controller.outputs.background-controller-digest }}"
```

Reply:
I don't think so, the job should have an output statement and other jobs should use that i think.

Reply:
I will fix it in a follow up PR.

Reply:
Are we trying to achieve something like the job that generates that digest value should replace that digest reference or what? The `value` is generating the digest using the steps mentioned in the action.yaml file right?